### PR TITLE
Trying out multiple sections

### PR DESCRIPTION
**Pull request by kneth**
I propose a new format of the changelog. I think it is easier to read when a release has multiple sections. The credit section can be used to acknowledge users who has contributed code or done testing using snapshot releases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
-* BREAKING CHANGE: DynamicRealm.executeTransaction() now throws IllegalArgumentException instead of silently accepting a null Transaction object.
-* BREAKING CHANGE: String setters now throw IllegalArgumentException instead of RealmError for invalid surrogates.
-* BREAKING CHANGE: DynamicRealm.distinct()/distinctAsync() and Realm.distinct()/distinctAsync() now throw IllegalArgumentException instead of UnsupportedOperationException for invalid type or unindexed field.
-* BREAKING CHANGE: All thread local change listeners are now delayed until the next Looper event instead of being triggered when committing.
-* BREAKING CHANGE: Removed RealmConfiguration.getSchemaMediator() from public API which was deprecated in 0.86.0. Please use RealmConfiguration.getRealmObjectClasses() to obtain the set of model classes (#1797).
-* BREAKING CHANGE: Realm.migrateRealm() throws a FileNotFoundException if the Realm file doesn't exist.
-* BREAKING CHANGE: It is now required to unsubscribe from all Realm RxJava observables in order to fully close the Realm (#2357).
+
+### Breaking changes
```
Comment (MGaetan89) [inline CHANGELOG.md]: Why not a `Breaking Changes` and a `Deprecations` sections to prevent repeating the first word?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
+
+* BREAKING: Realm has now to be installed as a Gradle plugin.
```
Comment (cmelchior) [inline CHANGELOG.md]: I would remove all the "BREAKING:" prefixes

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,36 +1,52 @@
...
+* BREAKING: Removed RealmConfiguration.getSchemaMediator() from public API which was deprecated in 0.86.0. Please use RealmConfiguration.getRealmObjectClasses() to obtain the set of model classes (#1797).
+* BREAKING: Realm.migrateRealm() throws a FileNotFoundException if the Realm file doesn't exist.
+* BREAKING: It is now required to unsubscribe from all Realm RxJava observables in order to fully close the Realm (#2357).
+* DEPRECATED: Realm.getInstance(Context). Use Realm.getInstance(RealmConfiguration) or Realm.getDefaultInstance() instead.
+* DEPRECATED: Realm.getTable(Class) which was public because of the old migration API. Use Realm.getSchema() or DynamicRealm.getSchema() instead.
+* DEPRECATED: Realm.executeTransaction(Transaction, Callback) and replaced it with Realm.executeTransactionAsync(Transaction), Realm.executeTransactionAsync(Transaction, OnSuccess), Realm.executeTransactionAsync(Transaction, OnError) and Realm.executeTransactionAsync(Transaction, OnSuccess, OnError).
+
+### Enhancements
+
 * Added support for custom methods, custom logic in accessors, custom accessor names, interface implementation and public fields in Realm objects. (#909)
```
Comment (cmelchior) [inline CHANGELOG.md]: I would remove all the "Added" prefixes as well. Bullets might need a slight rewrite for this though

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,36 +1,52 @@
...
 * Allowed "realm" and "row" as a field name of model class (#2255).
-* Updated Realm Core to 0.97.0
-  - Fixed bug when setting a null value to trigger RealmChangeListener (#2366).
-  - Fixed bug to prevent throwing BadVersionException (#2391).
+* RealmResults.size() now returns Integer.MAX_VALUE when actual size is greater than Integer.MAX_VALUE (#2129).
+* Removed allowBackup from AndroidManifest (#2307).
+
+### Bug fixes
+
+* Fixed an error occurring during test and connectedCheck of unit test example (#1934).
```
Comment (cmelchior) [inline CHANGELOG.md]: I would remove all the "Fixed" prefixes, same as above.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,36 +1,52 @@
...
+* Fixed bug in jsonExample (#2092).
+* Fixed bug when multiple calls of RealmResults.distinct() causes to return wrong results (#2198).
+* Fixed bug when calling DynamicRealmObject.setList() with RealmList<DynamicRealmObject> (#2368).
+* Fixed RealmChangeListeners not triggering correctly if findFirstAsync() didn't find any object. findFirstAsync() Observables now also correctly call onNext when the query completes in that case (#2200). (thank you @grahamsmith)
+
+### Updated Realm Core to 0.97.0
```
Comment (cmelchior) [inline CHANGELOG.md]: Not sure why this deserves a headline by itself. I would put it under the section where it fit the best, either "Enhancements" og "Bugfixes"

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,36 +1,52 @@
...
+
+* Fixed bug when setting a null value to trigger RealmChangeListener (#2366).
+* Fixed bug to prevent throwing BadVersionException (#2391).
+
+
+### Credits
+
+* Thanks to Bill West (@wmbest2) for snapshot testing.
```
Comment (cmelchior) [inline CHANGELOG.md]: 'grahamsmith' was not moved from the above.

**Comment by cmelchior** [comment]
I like the general idea of this.

File: CHANGELOG.md (hunk 5 above)
Comment (kneth) [inline CHANGELOG.md]: When we upgrade Realm Core, we could mention it before any subsection, and then simply add the items to either "Enhancements" or "Bug fixes".

Comment (cmelchior) [inline CHANGELOG.md]: You mean like 

```
### Enhancements
Upgraded Realm Core to 0.97
  - XXX

### Bug fixes
Upgraded Realm Core to 0.97
  - YYY
```

**Pull request merged**